addons/mrp/models/mrp_workorder.py

```diff
@@ -553,13 +553,15 @@ def _cal_cost(self):
             total += (duration / 60.0) * wo.workcenter_id.costs_hour
         return total
 
-    def button_start(self):
+    def button_start(self, from_action=False):
```

Comment:
Use a better name. Even if it's coming from action, you could use a name that define what it does. Something like raise_on_invalid_state